src/DataStructures/Tensor/Expressions/Contract.hpp

```diff
@@ -40,16 +40,12 @@ template <typename ReplacedArg1, typename ReplacedArg2, typename T, typename X,
           typename Symm, typename IndexList, typename Args>
 using ComputeContractedType = typename ComputeContractedTypeImpl<
     T, X,
-    tmpl::erase<tmpl::erase<Symm, tmpl::index_of<
-                                      Args, TensorIndex<ReplacedArg2::value>>>,
-                tmpl::index_of<Args, TensorIndex<ReplacedArg1::value>>>,
-    tmpl::erase<
-        tmpl::erase<IndexList,
-                    tmpl::index_of<Args, TensorIndex<ReplacedArg2::value>>>,
-        tmpl::index_of<Args, TensorIndex<ReplacedArg1::value>>>,
-    tmpl::erase<tmpl::erase<Args, tmpl::index_of<
-                                      Args, TensorIndex<ReplacedArg2::value>>>,
-                tmpl::index_of<Args, TensorIndex<ReplacedArg1::value>>>>::type;
+    tmpl::erase<tmpl::erase<Symm, tmpl::index_of<Args, ReplacedArg2>>,
```

Comment:
This would probably be a lot easier to read if you defined type aliases `contracted_symm`, `contracted_index_list` and `contracted_args` above